### ISSUE TITLE
Show optimal number of moves from the start of a new puzzle

### ISSUE THREAD

**Issue by aspiers** [supertorpe/chessendgametraining#14]
[Hi, I just discovered this awesome project!  Hope you don't mind if I submit a few suggestions :)  Here is the first one ...]

It only shows "You can reach mate in x" after the first move has been played.  This means it's impossible to tell whether your first move was a good one or a bad one.  It would be better if it showed it from the beginning.

**Comment by aspiers** [comment]
This also applies to already completed puzzles.  Currently I see something like "Personal record: 11 moves" but it would be great if it also showed "Best possible: 8 moves".

**Comment by supertorpe** [comment]
Suggestion are welcome. I have to decide whether to statically (at "compile" or "package" phase) calculate the smallest number of movements of each position, and store that data in the database, or calculate it at runtime (it would only be viable for syzygy positions).

**Comment by aspiers** [comment]
Yep, good point.  I guess statically would be better if possible, but either would work.
